www/react-base/src/data/ReactUtils.ts

```diff
@@ -0,0 +1,81 @@
+/*
+  This file is part of Buildbot.  Buildbot is free software: you can
+  redistribute it and/or modify it under the terms of the GNU General Public
+  License as published by the Free Software Foundation, version 2.
+
+  This program is distributed in the hope that it will be useful, but WITHOUT
+  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+  FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+  details.
+
+  You should have received a copy of the GNU General Public License along with
+  this program; if not, write to the Free Software Foundation, Inc., 51
+  Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+  Copyright Buildbot Team Members
+*/
+
+import {createContext, useContext, useEffect, useRef} from "react";
+import DataClient from "./DataClient";
+import {IDataAccessor} from "./DataAccessor";
+import {IDataCollection} from "./DataCollection";
+
+// The default value is not used as the context is injected
+export const DataClientContext =
+  createContext(new DataClient(undefined as any, undefined as any));
+
+export function useDataAccessor() {
```

Comment:
looks cool, but I think it deserves some documentation :)

Reply:
Agreed, this PR is bare minimum that's needed for the code to work _at all_. Documentation will come later :)